torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1305,7 +1305,18 @@ def _post_backward_hook(self, param: Parameter, *unused: Any) -> None:
                 if self.gradient_postdivide_factor > 1:
                     # Average grad by world_size for consistency with PyTorch DDP.
                     output.div_(self.gradient_postdivide_factor)
-                param.grad.data = output
+                accumulate_grad = getattr(param, "_saved_grad_shard", None) is not None
```

Comment:
Might be useful to add a small comment about how gradient accumulation without no_sync is implemented. From my understanding:

1. During backward, we point an attribute _saved_grad_shard to the gradient shard
2. If we are accumulating gradients, we accumulate it on _saved_grad_shard
3. When finalizing backwards before running the optimizer, we point p.grad to the saved grad shard so that the optimizer works on the right accumulated gradient.